scaleway/table_scaleway_kubernetes_cluster.go

```diff
@@ -0,0 +1,306 @@
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_cluster.listKubernetesClusters", "region_parsing_error", err)
+		return nil, err
+	}
+
+	quals := d.KeyColumnQuals
+	if quals["region"] != nil && quals["region"].GetStringValue() != region {
+		return nil, nil
+	}
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_cluster.listKubernetesClusters", "connection_error", err)
+		return nil, err
+	}
+
+	// Create SDK objects for Scaleway Kubernetes product
+	kubernetesApi := k8s.NewAPI(client)
+
+	req := &k8s.ListClustersRequest{
+		Region: parseRegionData,
+		Page:   scw.Int32Ptr(1),
+	}
+	// Additional filters
+	if quals["name"] != nil {
+		req.Name = scw.StringPtr(quals["name"].GetStringValue())
+	}
+
+	// Retrieve the list of clusters
+	maxResult := int64(100)
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < maxResult {
+			maxResult = *limit
+		}
+	}
+	req.PageSize = scw.Uint32Ptr(uint32(maxResult))
+
+	var count int
+
+	for {
+		resp, err := kubernetesApi.ListClusters(req)
+		if err != nil {
+			plugin.Logger(ctx).Error("scaleway_kubernetes_cluster.listKubernetesClusters", "query_error", err)
+			return nil, err
+		}
+
+		for _, cluster := range resp.Clusters {
+			d.StreamListItem(ctx, cluster)
+
+			// Increase the resource count by 1
+			count++
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		if resp.TotalCount == uint32(count) {
+			break
+		}
+		req.Page = scw.Int32Ptr(*req.Page + 1)
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getKubernetesCluster(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_cluster.getKubernetesCluster", "region_parsing_error", err)
+		return nil, err
+	}
+
+	if d.KeyColumnQuals["region"].GetStringValue() != region {
+		return nil, nil
+	}
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_cluster.getKubernetesCluster", "connection_error", err)
+		return nil, err
+	}
+
+	// Create SDK objects for Scaleway Kubernetes product
+	kubernetesApi := k8s.NewAPI(client)
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+	KubernetesRegion := d.KeyColumnQuals["region"].GetStringValue()
+
+	// No inputs
+	if id == "" && KubernetesRegion == "" {
+		return nil, nil
+	}
```

Comment:
```suggestion
	id := d.KeyColumnQuals["id"].GetStringValue()

	// No input id has been passed
	if id == "" {
		return nil, nil
	}
```

We don't need to check for `KubernetesRegion` since it is already being checked below - 
```
	if d.KeyColumnQuals["region"].GetStringValue() != region {
		return nil, nil
	}
```